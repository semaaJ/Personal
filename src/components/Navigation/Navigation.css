.navigation {
    display: flex;
    flex-direction: row;
    justify-content: center;
    list-style-type: none;
    list-style: none;
    align-items: center;
    padding: 0;
}

.navItem {
    margin: 10px;
    cursor: pointer;
}

.navItem slash {
    cursor: none;
}

.navItem a {
    text-decoration: none;
    font: 14px 'Inconsolata', monospace;
    text-transform: uppercase;
}

.underlined {
    color: #383737;
    font-size: 1em;
    text-decoration: none;
    background-position: 0 1.2em;
    background-size: 0 100%;
    background-repeat: no-repeat;
    transition: background .3s;
}

.underlined:hover {
    background-size: 100% 100%;
    background-image: linear-gradient(to right, #383737 0, #383737 100%);
}
