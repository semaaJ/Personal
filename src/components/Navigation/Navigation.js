import React from 'react';
import './Navigation.css';

const Navigation = () => {
    return (
        <div className="navigation">
                <div className="navItem underlined"><a>01. About</a></div>
                <div className="slash">//</div>
                <div className="navItem underlined"><a>02. Experience</a></div>
                <div className="slash">//</div>
                <div className="navItem underlined"><a>03. Projects</a></div>  
                <div className="slash">//</div>
                <div className="navItem underlined"><a>04. Contact</a></div> 
        </div>
    )
}   

export default Navigation;