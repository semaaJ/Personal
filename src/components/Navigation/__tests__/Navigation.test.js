import React from 'react';
import Navigation from '../Navigation';
import { shallow } from 'enzyme';
import '../../../__mocks__/intersectionObserverMock';
import "../../../setupTests"

test('should render Navigation correctly', () => {
    const component = shallow(<Navigation />);
    expect(component.html()).toMatchSnapshot();
});
