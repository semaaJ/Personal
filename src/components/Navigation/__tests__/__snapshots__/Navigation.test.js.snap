// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navigation correctly 1`] = `"<div class=\\"navigation\\"><div class=\\"navItem underlined\\"><a>01. About</a></div><div class=\\"slash\\">//</div><div class=\\"navItem underlined\\"><a>02. Experience</a></div><div class=\\"slash\\">//</div><div class=\\"navItem underlined\\"><a>03. Projects</a></div><div class=\\"slash\\">//</div><div class=\\"navItem underlined\\"><a>04. Contact</a></div></div>"`;
